pkg/tool/osdctl/osdctl.go

```diff
@@ -0,0 +1,191 @@
+package osdctl
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	gogithub "github.com/google/go-github/github"
+	"github.com/openshift/backplane-tools/pkg/source/github"
+)
+
+// Tool implements the interface to manage the 'osdctl' binary
+type Tool struct {
+	source *github.Source
+}
+
+func NewTool() *Tool {
+	t := &Tool{
+		source: github.NewGithubSource("openshift", "osdctl"),
+	}
+	return t
+}
+
+func (t *Tool) Name() string {
+	return "osdctl"
+}
+
+func (t *Tool) Install(rootDir, latestDir string) error {
+	// Pull latest release from GH
+	release, err := t.source.FetchLatestRelease()
+	if err != nil {
+		return err
+	}
+
+	// Determine which assets to download
+	var checksumAsset gogithub.ReleaseAsset
+	var osdctlBinaryAsset gogithub.ReleaseAsset
+	var arch = runtime.GOARCH
+	if arch == "amd64" {
+		arch = "x86_64"
+	}
+	for _, asset := range release.Assets {
+		fmt.Println(asset.GetName())
+
+		if asset.GetName() == "sha256sum.txt" {
+			if checksumAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate osdctl checksum assets")
+			}
+			checksumAsset = asset
+			if osdctlBinaryAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate osdctl binary assets")
+			}
+			osdctlBinaryAsset = asset
+			continue
+		}
+	}
+	// Ensure both checksum and binary were retrieved
+	if checksumAsset.GetName() == "" || osdctlBinaryAsset.GetName() == "" {
+		return fmt.Errorf("failed to find osdctl or it's checksum")
+	}
+
+	// Download the arch- & os-specific assets
+	toolDir := t.toolDir(rootDir)
+	versionedDir := filepath.Join(toolDir, release.GetTagName())
+	err = os.MkdirAll(versionedDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("failed to create version-specific directory '%s': %w", versionedDir, err)
+	}
+
+	err = t.source.DownloadReleaseAssets([]gogithub.ReleaseAsset{checksumAsset, osdctlBinaryAsset}, versionedDir)
+	if err != nil {
+		return fmt.Errorf("failed to download one or more assets: %w", err)
+	}
+
+	// Verify checksum of downloaded assets
+	osdctlBinaryFilepath := filepath.Join(versionedDir, osdctlBinaryAsset.GetName())
+	fileBytes, err := os.ReadFile(osdctlBinaryFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl binary file '%s' while generating sha256sum: %w", osdctlBinaryFilepath, err)
+	}
+	sumBytes := sha256.Sum256(fileBytes)
+	// TODO - there's probably a better way to do this
+	binarySum := fmt.Sprintf("%x", sumBytes[:])
+
+	checksumFilePath := filepath.Join(versionedDir, checksumAsset.GetName())
+	checksumBytes, err := os.ReadFile(checksumFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl checksum file '%s': %w", checksumFilePath, err)
+	}
+	checksum := strings.Split(string(checksumBytes), " ")[0]
```

Comment:
This doesn't look like it'd return the correct checksum value for most distros. [osdctl's sha256sum.txt](https://github.com/openshift/osdctl/releases/download/v0.14.4/sha256sum.txt) file is a list containing the checksum for every asset. We need to determine which checksum to use based on the name of the `osdctlBinaryAsset`